validator/src/server/cli.rs

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use clap::{App, Arg, ArgMatches};
+
+const DISTRIBUTION_NAME: &str = "sawtooth-validator";
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+pub fn get_cli_args() -> String {
+    let matches = parse_args();
+
+    let (bind_component, bind_network) = parse_bindings(&matches);
+
+    json!({
+        "bind_component": bind_component,
+        "bind_network": bind_network,
+        "config_dir": matches.value_of("config_dir"),
+        "endpoint": matches.value_of("endpoint"),
+        "maximum_peer_connectivity": matches.value_of("maximum_peer_connectivity"),
+        "minimum_peer_connectivity": matches.value_of("minimum-peer-connectivity"),
+        "roles": match matches.value_of("network_auth") {
+            Some(network_auth) => Some(json!({"network": network_auth})),
+            None => None,
+        },
+        "opentsdb_db": matches.value_of("opentsdb-db"),
+        "opentsdb_url": matches.value_of("opentsdb-url"),
+        "peering": matches.value_of("peering"),
+        "peers": parse_comma_separated_args("peers", &matches),
+        "scheduler": matches.value_of("scheduler"),
+        "seeds": parse_comma_separated_args("seeds", &matches),
+        "verbose": matches.value_of("verbose"),
+    }).to_string()
+}
+
+fn parse_args<'a>() -> ArgMatches<'a> {
+    let app = App::new(DISTRIBUTION_NAME)
```

Comment:
Any reason not to use the macro syntax? It is much more concise.

Reply:
The macro syntax does not support `possible_values`